File: 07_LLD Case Studies/Car Parking Lot System/src/models/AdminAccount.java
package models;

import Interface.Account;
import Interface.ParkingSpot;

public class AdminAccount extends Account {
        public boolean addParkingSpot(ParkingSpot spot){
            return false;
        }

        public boolean addEntrance(Enterance entry){
            return false;
        }

        public boolean addExit(Exit exit){
            return false;
        }

        public boolean addDisplayBoard(DisplayBoard board){
            return false;
        }

        @Override
        public boolean resetPassword(){
            return false;
        }
}

File: 09_Design Problems/01_Parking Lot System/Essentials/AdminAccount.cpp
#include <iostream>
#include "../Interfaces/Account.cpp"
#include "../Interfaces/ParkingSpot.cpp"
#include "../Essentials/Entrance.cpp"
#include "../Essentials/Exit.cpp"
#include "../Essentials/DisplayBoard.cpp"

using namespace std;

class AdminAccount : public IAccount {
    
    public:
        bool resetPassword() override {
            cout << "Admin Password Reset" << endl;
            return false; 
        }
        bool AddParkingSpot(IParkingSpot* spot) {
            cout << "Parking Spot Added" << endl;
            return true; 
        } 
        bool AddEntrance(Entrance* entry) {
            cout << "Entrance Added" << endl;
            return true; 
        }
        bool AddExit(Exit* exit) {
            cout << "Exit Added" << endl;
            return true; 
        }
        bool AddDisplayBoard(DisplayBoard* board){
            cout << "Display Board Added" << endl;
            return true;
        }
};